src/types/colorTable.cpp

```diff
@@ -1044,23 +635,57 @@ try
             return std::nullopt;
         }
 
+        if (c >= L'0' && c <= L'9')
+        {
+            foundVariant = true;
+            variantIndex *= 10;
+            variantIndex += c - L'0';
+            continue;
+        }
+
         // Ignore spaces.
         if (std::iswspace(c))
         {
             continue;
         }
 
+        if (foundVariant)
+        {
+            // Variant should be at the end of the string, e.g., "yellow3".
+            // This means another alphabet character is seen, e.g., "yellow3a".
+            // This is invalid so return early.
+            return std::nullopt;
+        }
+
         ss << gsl::narrow_cast<char>(std::towlower(c));
     }
 
     std::string name(ss.str());
-    const auto iter = xorgAppColorTable.find(name);
-    if (iter == xorgAppColorTable.end())
+    const auto variantColorIter = xorgAppVariantColorTable.find(name);
+    if (variantColorIter != xorgAppVariantColorTable.end())
     {
-        return std::nullopt;
+        const auto colors = variantColorIter->second;
+        if (variantIndex < colors.size())
+        {
+            return colors.at(variantIndex);
+        }
+    }
+
+    const auto colorIter = xorgAppColorTable.find(name);
+    if (colorIter != xorgAppColorTable.end())
+    {
+        return colorIter->second;
     }
 
-    return iter->second;
+    if (name == "gray" || name == "grey")

```

Comment:
OK this is kinda stupid but it works pretty well. The “gray" series and "grey" series of color are identical. This saves us a lot of space.